airflow/api_fastapi/execution_api/routes/variables.py

```diff
@@ -67,6 +67,20 @@ def get_variable(variable_key: str, token: deps.TokenDep) -> VariableResponse:
     return VariableResponse(key=variable_key, value=variable_value)
 
 
+@router.post(
+    "",
+    responses={
+        status.HTTP_401_UNAUTHORIZED: {"description": "Unauthorized"},
+        status.HTTP_403_FORBIDDEN: {"description": "Task does not have access to the variable"},
+    },
+)
+def post_variable(body: VariablePostBody, token: deps.TokenDep) -> VariableResponse:
```

Comment:
Design question: Should this be `POST /execution/variables` with key in the body, or should it perhaps be `PUT /execution/variables/{variable_key}".

Possible advantage of the second approach: it shows in the request logs when a specific variable is updated which might be nice.

Ultimately it doesn't really matter, and there is no "correct" way, it's just up to us.

Reply:
Right.

Two reasons i chose POST `/execution/variables` over the other one:
1. `variable_key` and `key` again in the payload didn't feel too right as it was avoidable duplication.
2. We have to handle cases where `variable_key` in the parameters and `key` in the payload are potentially different. Could be quite chaotic!

Also, legacy implements it this way!

Reply:
On 1. Yeah, if it's in the URL we wouldn't have it in the body too, just value and desc.

Reply:
I see, it doesn't really matter to me which way we go then. Just a matter of agreement with everyone

Reply:
Based on offline discussion we decided to go with `PUT /execution/variables/{variable_key}`